tendermint-rs/src/amino_types/block_id.rs

```diff
@@ -18,6 +19,19 @@ impl block::ParseId for BlockId {
     }
 }
 
+impl ConsensusMessage for BlockId {
+    fn validate_basic(&self) -> Result<(), ValidationError> {
+        // Hash can be empty in case of POLBlockID in Proposal.
+        if !self.hash.is_empty() && self.hash.len() != SHA256_HASH_SIZE {
+            return Err(ValidationError::new(ValidationErrorKind::InvalidHashSize));
+        }
+        match self.parts_header {
+            Some(ref psh) => psh.validate_basic(),
+            None => Ok(()), // TODO: is an empty PartsSetHeader really OK here?
```

Comment:
It's probably an error if the partsset header is missing completely (although it might be "empty")?
/cc @ebuchman  @ValarDragon @melekes

In that case this line would change to:
```rust self.parts_header.as_ref().map_or(Err(ValidationErrorKind::MissingPartsSetHeader.into()), |psh| psh.validate_basic())
```  
or so. In that case the match might make more sense.